### ISSUE TITLE
Questions about training and Environment Setup

### ISSUE THREAD

**Issue by gst0312** [TempleRAIL/drl_vo_nav#40]
Hi, thank you for such great work!
I have encountered 2 questions:
Question 1: Mean Rewards Not Improving in Training
I trained the model for over 800,000 steps, but I noticed that the mean rewards have not improved and are oscillating. Interestingly, the initial mean rewards were already greater than 0, but the success rate remained low. After testing, the final success rate was only below 30%.
The only modification I made to the code was changing the batch size from 128 to 32 due to GPU memory limitations.
Could this change in batch size cause such a significant impact on the training results? Or is there something else I might be missing that could explain the lack of improvement?

<img width="617" alt="Image" src="https://github.com/user-attachments/assets/405f1129-e888-4e55-8e00-b5b93eb31639" />

Also I forgot to screen shot the losses, but they both went up pretty quickly and started to oscillate.

Question 2: Error When Using Singularity on Ubuntu 20.04.02
I attempted to run the setup on Ubuntu 20.04.02 using Singularity, but encountered the following error:

<img width="1459" alt="Image" src="https://github.com/user-attachments/assets/ee6f1173-4f36-44e4-b83b-63ad70401de8" />

I’m unsure how to resolve this issue. Could this be related to Singularity's GPU support or specific OpenGL/GLX compatibility issues?

**Comment by zzuxzt** [comment]
Thanks for your interest in our work.
For question 1, you can try to train it for a much longer time (more than 1M). The initial reward is greater than 0 because I provide a pre-trained model to speed up training. However, tuning parameters like batch size may also cause issues, as DRL training is a bit unstable, please refer to https://github.com/TempleRAIL/drl_vo_nav/issues/39#issuecomment-2609386810 and https://github.com/TempleRAIL/drl_vo_nav/issues/25#issuecomment-2087934233 for more training experience. Also, please note that this open source version code directly uses accurate pedestrian information from the Gazebo simulator instead of using the YOLO&MHT pipeline as in our paper (due to some commercial library licensing restrictions in the MHT tracker), which results in changes in the training environment. I cannot be sure whether it can be 100% reproduced for training as I do not have time to test it.

For question 2, you can try to upgrade the request package:
```
pip3 install --upgrade requests
```
